libs/x-content/src/main/java/org/elasticsearch/xcontent/XContentParserConfiguration.java

```diff
@@ -99,10 +103,15 @@ public XContentParserConfiguration withFiltering(Set<String> includeStrings, Set
             deprecationHandler,
             restApiVersion,
             FilterPath.compile(includeStrings),
-            FilterPath.compile(excludeStrings)
+            FilterPath.compile(excludeStrings),
+            supportDotInFieldName
         );
     }
 
+    public XContentParserConfiguration withSupportDotInFieldName(boolean supportDotInFieldName) {
+        return new XContentParserConfiguration(registry, deprecationHandler, restApiVersion, includes, excludes, supportDotInFieldName);
+    }
```

Comment:
I like the name I used for this variable more. And I like forcing folks to configure it when configuring filtering. That way then they build the filter they have to think about it.

Reply:
I like yours too. Forgive my poor Englisth, LOL